### PR TITLE
change subtitle on empty state + marketpalce recycler empty view crash

### PR DESCRIPTION
**Pull request by OrenZak**
- Change subtitle on empty state to something concrete.
- Change all the observable data to post the values on Main Thread instead of set on the same thread
- Fix recycler crash on, updating item after empty state by setting the index of the offer to be 1 instead of 0.

### CODE REVIEW

File: kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/marketplace/presenter/MarketplacePresenter.java [hunk 1]
```diff
@@ -46,6 +52,12 @@
 
 	private Observer<Offer> pendingOfferObserver;
 	private Observer<Order> completedOrderObserver;
+
+	private ObservableData<Boolean> isEarnEmpty = ObservableData.create();
+	private ObservableData<Boolean> isSpendEmpty = ObservableData.create();
```
Comment (YohayBar) [inline kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/marketplace/presenter/MarketplacePresenter.java]: default value?

File: kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/marketplace/presenter/MarketplacePresenter.java [hunk 2]
```diff
@@ -36,6 +38,10 @@
 
 	private static final int NOT_FOUND = -1;
 
+	private static final String EMPTY_SUBTITLE = "Check back tomorrow for more great opportunities";
+	private static final String EARN_SUBTITLE = "Complete tasks and earn KIN";
```
Comment (YohayBar) [inline kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/marketplace/presenter/MarketplacePresenter.java]: String.xml please

File: kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/marketplace/presenter/MarketplacePresenter.java (hunk 1 above)
Comment (OrenZak) [reply kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/marketplace/presenter/MarketplacePresenter.java]: shouldn't be, should be set on the first time we are setting offers

File: kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/marketplace/presenter/MarketplacePresenter.java [hunk 3]
```diff
@@ -118,51 +157,57 @@ private void removeOfferFromList(Offer offer) {
...
+			view.notifySpendItemInserted(getSafeIndexSpendEmptyState(index));
 		}
 	}
 
+	private int getSafeIndexEarnEmptyState(final int index) {
+		return isEarnEmpty.getValue() ? 1 : index;
+	}
+
+	private int getSafeIndexSpendEmptyState(final int index) {
+		return isSpendEmpty.getValue() ? 1 : index;
```
Comment (YohayBar) [inline kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/marketplace/presenter/MarketplacePresenter.java]: ?

File: kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/marketplace/presenter/MarketplacePresenter.java [hunk 4]
```diff
@@ -118,51 +157,57 @@ private void removeOfferFromList(Offer offer) {
 
 	private void setEarnEmptyViewIfNeeded() {
 		if (earnList.size() == 0) {
-			if (view != null) {
-				view.setEarnEmptyView();
-			}
+			isEarnEmpty.postValue(true);
```
Comment (YohayBar) [inline kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/marketplace/presenter/MarketplacePresenter.java]: ?

File: kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/marketplace/presenter/MarketplacePresenter.java [hunk 5]
```diff
@@ -118,51 +157,57 @@ private void removeOfferFromList(Offer offer) {
...
 		}
 	}
 
 	private void setSpendEmptyViewIfNeeded() {
 		if (spendList.size() == 0) {
-			if (view != null) {
-				view.setSpendEmptyView();
-			}
+			isSpendEmpty.postValue(true);
```
Comment (YohayBar) [inline kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/marketplace/presenter/MarketplacePresenter.java]: ?

File: kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/marketplace/presenter/MarketplacePresenter.java [hunk 6]
```diff
@@ -152,10 +158,18 @@ private void notifySpendItemRemoved(int index) {
 
 	private void notifySpendItemInserted(int index) {
 		if (view != null) {
-			view.notifySpendItemInserted(index);
+			view.notifySpendItemInserted(getSafeIndexSpendEmptyState(index));
 		}
 	}
 
+	private int getSafeIndexEarnEmptyState(final int index) {
+		return isEarnListEmpty? 1 : index;
```
Comment (OrenZak) [inline kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/marketplace/presenter/MarketplacePresenter.java]: format

File: kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/marketplace/presenter/MarketplacePresenter.java [hunk 7]
```diff
@@ -7,6 +7,7 @@
 import com.google.gson.JsonSyntaxException;
 import com.kin.ecosystem.KinCallback;
 import com.kin.ecosystem.base.BasePresenter;
+import com.kin.ecosystem.base.ObservableData;
```
Comment (OrenZak) [inline kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/marketplace/presenter/MarketplacePresenter.java]: remove

File: kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/marketplace/view/MarketplaceActivity.java [hunk 8]
```diff
@@ -29,12 +30,12 @@
 
 public class MarketplaceActivity extends BaseToolbarActivity implements IMarketplaceView {
 
-    private IMarketplacePresenter marketplacePresenter;
+	private IMarketplacePresenter marketplacePresenter;
```
Comment (OrenZak) [inline kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/marketplace/view/MarketplaceActivity.java]: format

**Pull request merged**